src/dataframe/dataframe.jl

```diff
@@ -165,9 +165,11 @@ julia> DataFrame([1 0; 2 0], :auto) # matrix constructor
    2 │     2      0
 ```
 """
-struct DataFrame <: AbstractDataFrame
+mutable struct DataFrame <: AbstractDataFrame
     columns::Vector{AbstractVector}
     colindex::Index
+    metadata::Union{Nothing, Dict{String, Any}}
+    column_metadata::Union{Nothing, Dict{Int, Dict{String, Any}}}
```

Comment:
For consistency with `colindex`:
```suggestion
    colmetadata::Union{Nothing, Dict{Int, Dict{String, Any}}}
```